k8s/pkg/registryserver/nse.go

```diff
@@ -52,11 +52,22 @@ func (rs *nseRegistryService) RegisterNSE(ctx context.Context, request *registry
 			logrus.Errorf("Failed to register nsm: %s", err)
 			return nil, err
 		}
-		nseResponse, err := rs.cache.AddNetworkServiceEndpoint(&v1.NetworkServiceEndpoint{
-			ObjectMeta: metav1.ObjectMeta{
+
+		var objectMeta metav1.ObjectMeta
+		if request.GetNetworkserviceEndpoint().EndpointName == "" {
+			objectMeta = metav1.ObjectMeta{
 				GenerateName: request.GetNetworkService().GetName(),
 				Labels:       labels,
-			},
+			}
+		} else {
+			objectMeta = metav1.ObjectMeta{
+				Name: request.GetNetworkserviceEndpoint().EndpointName,
+				Labels:       labels,
+			}
+		}
+
+		nseResponse, err := rs.cache.AddNetworkServiceEndpoint(&v1.NetworkServiceEndpoint{
```

Comment:
We should not send error in case same NSE is already exists. Could you please add tests for this case. Thanks.

Reply:
Method `cache.AddNetworkServiceEndpoint` already implements logic of ignoring this error:
```go
func (rc *registryCacheImpl) AddNetworkServiceEndpoint(nse *v1.NetworkServiceEndpoint) (*v1.NetworkServiceEndpoint, error) {
	if existingNse := rc.networkServiceEndpointCache.Get(nse.GetName()); existingNse != nil {
		return existingNse, nil
	}

	nseResponse, err := rc.clientset.NetworkservicemeshV1().NetworkServiceEndpoints(rc.nsmNamespace).Create(nse)
	if err == nil {
		rc.networkServiceEndpointCache.Add(nseResponse)
		return nseResponse, nil
	}

	if apierrors.IsAlreadyExists(err) {
		return nse, nil
	}

	return nil, err
}
```

Reply:
But actually if we receive `AlreadyExists` we should check equality of `NsmName` fields. And in case they are not equal we need to return error and ensure that healing failed and everything correctly closed